compiler-cli/src/hex/auth.rs

```diff
@@ -127,26 +143,43 @@ encrypt your Hex API key.
                 detail: e.to_string(),
             })?;
 
-        Ok(Some(UnencryptedApiKey { unencrypted }))
+        Ok(Some(UnencryptedApiKey {
+            unencrypted,
+            username,
+        }))
     }
 
     pub fn read_stored_api_key(&self) -> Result<Option<EncryptedApiKey>> {
         let path = global_hexpm_credentials_path();
         if !path.exists() {
             return Ok(None);
         }
-        let text = crate::fs::read(&path)?;
-        let mut chunks = text.splitn(2, '\n');
-        let Some(name) = chunks.next() else {
-            return Ok(None);
-        };
-        let Some(encrypted) = chunks.next() else {
-            return Ok(None);
-        };
-        Ok(Some(EncryptedApiKey {
-            name: name.to_string(),
-            encrypted: encrypted.to_string(),
-        }))
+
+        let encrypted_key = EncryptedApiKey::load(&path);
+
+        if let Ok(encrypted_key) = encrypted_key {
```

Comment:
This silently discards file system errors!

Let's move all of this loading logic into the load function, and have it read once, return an error if the file isn't readable, and also if the file is in neither format.

Reply:
Good call, than you for your feedback.

I moved the logic to a separate function as suggested, it does also read better.